src/transaction/Transaction.js

```diff
@@ -1745,6 +1823,99 @@ export default class Transaction extends Executable {
+                    );
+
+                    if (shouldRemove) {
+                        if (sigPair.ed25519) {
+                            removedSignatures.push(sigPair.ed25519);
+                        } else if (sigPair.ECDSASecp256k1) {
+                            removedSignatures.push(sigPair.ECDSASecp256k1);
+                        }
+                    }
+
+                    return !shouldRemove;
+                },
+            );
+        }
+
+        return removedSignatures;
+    }
+
+    /**
+     * Determines whether a signature should be removed based on the provided public key.
+     *
+     * @param {HashgraphProto.proto.ISignaturePair} sigPair - The signature pair object that contains
+     *        the public key prefix and signature to be evaluated.
+     * @param {string} publicKeyHex - The hexadecimal representation of the public key to compare against.
+     * @returns {boolean} `true` if the public key prefix in the signature pair matches the provided public key,
+     *          indicating that the signature should be removed; otherwise, `false`.
+     */
+    _shouldRemoveSignature = (sigPair, publicKeyHex) => {
+        const sigPairPublicKeyHex = hex.encode(
+            sigPair?.pubKeyPrefix || new Uint8Array(),
+        );
+
+        const matchesPublicKey = sigPairPublicKeyHex === publicKeyHex;
+
+        return matchesPublicKey;
+    };
+
+    /**
+     * Collects all signatures from signed transactions and returns them in a format keyed by public key.
+     *
+     * @returns {{ [publicKey: PublicKey]: Uint8Array[] }} The collected signatures keyed by public key.
+     */
+    _collectSignaturesByPublicKey() {
+        /** @type {{ [publicKey: string]: Uint8Array[] }} */
+        const collectedSignatures = {};
+
+        // Iterate over the signed transactions and collect signatures
+        for (const transaction of this._signedTransactions.list) {
+            // Collect the signatures
+            if (transaction.sigMap && transaction.sigMap.sigPair) {
+                for (const sigPair of transaction.sigMap.sigPair) {
+                    let signature;
+                    if (sigPair.ed25519) {
+                        signature = sigPair.ed25519;
+                    } else if (sigPair.ECDSASecp256k1) {
+                        signature = sigPair.ECDSASecp256k1;
+                    }
+
+                    if (signature && sigPair.pubKeyPrefix) {
+                        const publicKeyHex = hex.encode(sigPair.pubKeyPrefix);
+
+                        // Initialize the structure for this publicKey if it doesn't exist
+                        if (!collectedSignatures[publicKeyHex]) {
+                            collectedSignatures[publicKeyHex] = [];
+                        }
+
+                        // Add the signature to the corresponding public key
+                        collectedSignatures[publicKeyHex].push(signature);
+                    }
+                }
+            }
```

Comment:
Same as above. Please:
- Use the logical OR ( || ) to determine the signature pair.
- `return []` if there is no sigmap or sigpair.
-  also `for(const {sigPair} ... )`
- `if (!signature && !sigPair.pubKeyPrefix) return;`

Now that I think of it maybe we should return empty arrays, `undefined`, `null` but actually throw an error it might help with future debugging.

Reply:
I don't think this is actually an error. If you run the collect method on a transaction without signatures, it would be expected to return an empty array if there are no signatures present.